dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/ClientBase.cs

```diff
@@ -85,7 +85,7 @@ protected static T DeserializeResponse<T>(string body)
         }
         catch (JsonException exc)
         {
-            throw new KernelException("Unexpected response from model", exc)
+            throw new KernelException("Unexpected response from model")
```

Comment:
Please, pass the inner exception, is a good practice 👍🏻

Reply:
But you don't want to expose details of Json validation like we talked on discord. or I misunderstood that.

Reply:
Was a misunderstand. I can clarify.

Before if we had an exception it would be on a 200 OK response, which turns out to be something expected. 

Now that the parts is optional, that exception will not happen and we will be compatible to that result.

With that is ok to have the JsonException passed as InnerException for things that are indeed unexpected.